site/src/pages/TerminalPage/TerminalPage.tsx

```diff
@@ -470,21 +470,62 @@ const useReloading = (isDisconnected: boolean) => {
 
   // Retry connection on key press when it is disconnected
   useEffect(() => {
-    if (!isDisconnected) {
+    if (!isDisconnected || status === "reloading") {
       return;
     }
 
-    const keyDownHandler = () => {
-      setStatus("reloading");
-      window.location.reload();
+    // Keep track of modifier keys since we want to avoid reconnecting while
+    // modifiers are held.  This covers cases where the terminal unexpectedly
+    // tries to reconnect like when pressing ctrl+w, ctrl+r, and so on.  This
+    // will not work if you pressed a modifier before the disconnect and are
+    // still holding it; if we need to account for that we will need to listen
+    // for modifier keys while connected as well.
+    const modifierKeyState: Record<string, boolean> = {
+      Alt: false,
+      AltGraph: false,
+      CapsLock: false,
+      Control: false,
+      Fn: false,
+      FnLock: false,
+      Meta: false,
+      NumLock: false,
+      ScrollLock: false,
+      Shift: false,
+      Symbol: false,
+      SymbolLock: false,
     };
 
-    document.addEventListener("keydown", keyDownHandler);
+    const isModifier = (event: KeyboardEvent): boolean => {
+      return typeof modifierKeyState[event.key] !== "undefined";
+    };
+
+    const isModified = (): boolean => {
+      return Object.values(modifierKeyState).includes(true);
+    };
+
+    const keyDownHandler = (event: KeyboardEvent) => {
+      if (isModifier(event)) {
+        modifierKeyState[event.key] = true;
+      } else if (!isModified()) {
+        setStatus("reloading");
```

Comment:
Yeah I thought the same thing, it could depend on the browser or something, but in Chromium I tested via a `console.log` statement and it will happily run multiple times until the reload "kicks in".

Reply:
That said, running `window.location.reload()` multiple times is probably harmless.

Reply:
One more thing, setting the status here to `reloading` causes the overlay to display "Reloading...", so whatever `window.location.reload()` is doing, it appears to not quite be instant.

Reply:
wow, ok 😅 I thought I'd even played with this before and touching `location` instantly killed your code

Reply:
that said, it seems like this would benefit from being instant, rather than waiting for react to do an update cycle. maybe we should use `useRef` for this?

Reply:
lol I know what you mean, I could have sworn I did the same!

Reply:
Oh right, good point.  Maybe it would be nice if the UI could update immediately as well since if the reload happens too fast they might never see the `Reconnecting...` text?  Wonder if we can force an update or something.

Reply:
I experimented a bit to see if I could get the keydown handler firing twice before React updates and the listener is removed but apparently I am not able to mash a key fast enough lol

I did find a bug though, if you ctrl+tab away then come back via click, the page still thinks ctrl is being pressed and nothing will reload until you press and release ctrl again.  Not sure if this can be solved with keydown...too bad it is not like the deprecated `keypress` event that would tell you if modifier keys were being held down.

Reply:
Wait what no `keydown` does have that information.  What am I even doing??